### PR TITLE
#81-create-main-operation-with-users-via-mediatr

### PR DESCRIPTION
**Pull request by SokhibJones**
Added CRUD operations using CQRS pattern for User model via MediatR

### CODE REVIEW

File: src/BudgetManager/BudgetManager.CQRS/Handlers/UserHandlers/AddUserHandler.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        public AddUserHandler(IBaseRepository<User> userRepository, IMapper mapper)
+        {
+            _userRepository = userRepository;
+            _mapper = mapper;
+        }
+
+        public async Task<Guid> Handle(AddUserCommand request, CancellationToken cancellationToken)
+        {
+            AddUserDTO requestUser = request.user;
+            User mappedUser = _mapper.Map<User>(requestUser);
```
Comment (Saigunas) [inline src/BudgetManager/BudgetManager.CQRS/Handlers/UserHandlers/AddUserHandler.cs:24]: This is unnecessary.

**Comment by Saigunas** [comment]
You should also implement controller.

File: src/BudgetManager/BudgetManager.CQRS/Commands/UserCommands/UpdateUserCommand.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿using BudgetManager.Model;
+using BudgetManager.SDK.DTO.UserDTOs;
+using MediatR;
+
+namespace BudgetManager.CQRS.Commands.UserCommands
+{
+    public record UpdateUserCommand(UpdateUserDTO updateUser) : IRequest<User>;
```
Comment (VladislavGromadtsov) [inline src/BudgetManager/BudgetManager.CQRS/Commands/UserCommands/UpdateUserCommand.cs]: Here should be UserResponse

File: src/BudgetManager/BudgetManager.CQRS/Handlers/UserHandlers/UpdateUserHandler.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            var existingUser = await _userRepository.FindByIdAsync(updateUser.Id, cancellationToken);
+            if (existingUser is null)
+            {
+                return null;
+            }
+
+            var mappedUser = _mapper.Map<User>(updateUser);
+            var response = await _userRepository.ReplaceOneAsync(mappedUser, cancellationToken);
+
+            return response;
```
Comment (VladislavGromadtsov) [inline src/BudgetManager/BudgetManager.CQRS/Handlers/UserHandlers/UpdateUserHandler.cs]: This method should have only one call to database. In this case you make 2

**Review by VladislavGromadtsov — CHANGES_REQUESTED** [review]

**Review by VladislavGromadtsov — APPROVED** [review]

**Review by paulius27 — APPROVED** [review]

**Pull request merged**